orchagent/mirrororch.cpp

```diff
@@ -562,175 +601,165 @@ bool MirrorOrch::updateSessionDstPort(const string& name, MirrorEntry& session)
 
     assert(session.sessionId != SAI_NULL_OBJECT_ID);
 
-    SWSS_LOG_INFO("Updating mirror session %s destination port\n", name.c_str());
+    Port port;
+    m_portsOrch->getPort(session.neighborInfo.portId, port);
 
-    sai_status_t status = SAI_STATUS_SUCCESS;
     sai_attribute_t attr;
-
     attr.id =  SAI_MIRROR_SESSION_ATTR_MONITOR_PORT;
     attr.value.oid = session.neighborInfo.portId;
 
-    status = sai_mirror_api->set_mirror_session_attribute(session.sessionId, &attr);
+    sai_status_t status = sai_mirror_api->
+        set_mirror_session_attribute(session.sessionId, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to update mirror session %s monitor port to %s, rv:%d",
+                name.c_str(), port.m_alias.c_str(), status);
+        return false;
+    }
+
+    SWSS_LOG_NOTICE("Update mirror session %s monitor port to %s",
+            name.c_str(), port.m_alias.c_str());
+
+
+    return true;
+}
+
+bool MirrorOrch::updateSessionVlan(const string& name, MirrorEntry& session)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+    vector<sai_attribute_t> attrs;
+
+    if (session.neighborInfo.port.m_type == Port::VLAN)
+    {
+        attr.id = SAI_MIRROR_SESSION_ATTR_VLAN_HEADER_VALID;
+        attr.value.booldata = true;
+        attrs.push_back(attr);
+
+        attr.id =SAI_MIRROR_SESSION_ATTR_VLAN_ID;
+        attr.value.u16 = session.neighborInfo.port.m_vlan_info.vlan_id;
+        attrs.push_back(attr);
+    }
+    else
+    {
+        attr.id = SAI_MIRROR_SESSION_ATTR_VLAN_HEADER_VALID;
+        attr.value.booldata = false;
+        attrs.push_back(attr);
+    }
+
+    sai_status_t status = sai_mirror_api->
+        set_mirror_session_attribute(session.sessionId, &attr);
```

Comment:
>attr [](start = 57, length = 4)

this is not correct. you need to iterate all attributes and update one-by-one.

Reply:
I'll fix this